packages/primeng/src/scroller/scroller.ts

```diff
@@ -546,12 +551,20 @@ export class Scroller extends BaseComponent implements OnInit, AfterContentInit,
 
     get loadedColumns() {
         if (this._columns && (this.both || this.horizontal)) {
-            return this.d_loading && this._loaderDisabled ? (this.both ? this.loaderArr[0] : this.loaderArr) : this._columns.slice(this.both ? this.first.cols : this.first, this.both ? this.last.cols : this.last);
+            return this.d_loading && this._loaderDisabled ? (this.both ? this.loaderArr[0] : this.loaderArr) : this._columns.slice(this._first.main, this._last.main);
         }
 
         return this._columns;
     }
 
+    private get _first(): GridItem {
+        return { main: this.first?.[this.horizontal ? 'cols' : 'rows'] ?? this.first, cross: this.horizontal ? 0 : (this.first?.['cols'] ?? 0) };
```

Comment:
Thinking in terms of `main` and `cross` axes (similar to CSS flexbox flow model)  makes it easier to deal with than `rows` and `cols`, makes code more monomorphic, less `if` statements to check which `orientation` it is now to know whether we should use `rows` or `cols`.

I would change the input `first` and `last` fields to that model, but it would introduce even more breaking changes, and I don't know if that is what you want.